test/e2e/virtual/apiexport/authorizer_test.go

```diff
@@ -346,3 +354,175 @@ func TestAPIExportAuthorizers(t *testing.T) {
 	require.Error(t, err, "expected error, got none")
 	require.True(t, errors.IsNotFound(err))
 }
+
+func TestRootAPIExportAuthorizers(t *testing.T) {
+	t.Parallel()
+	framework.Suite(t, "control-plane")
+
+	server := framework.SharedKcpServer(t)
+
+	ctx, cancel := context.WithCancel(context.Background())
+	t.Cleanup(cancel)
+
+	org := framework.NewOrganizationFixture(t, server)
+
+	serviceWorkspace := framework.NewWorkspaceFixture(t, server, org, framework.WithName("provider"))
+	userWorkspace := framework.NewWorkspaceFixture(t, server, org, framework.WithName("consumer"))
+
+	cfg := server.BaseConfig(t)
+
+	kubeClient, err := kcpkubernetesclientset.NewForConfig(rest.CopyConfig(cfg))
+	require.NoError(t, err)
+	kcpClient, err := kcpclientset.NewForConfig(rest.CopyConfig(cfg))
+	require.NoError(t, err)
+
+	providerUser := "user-1"
+	consumerUser := "user-2"
+
+	framework.AdmitWorkspaceAccess(t, ctx, kubeClient, org, []string{providerUser, consumerUser}, nil, []string{"access"})
+	framework.AdmitWorkspaceAccess(t, ctx, kubeClient, serviceWorkspace, []string{providerUser}, nil, []string{"admin", "access"})
+	framework.AdmitWorkspaceAccess(t, ctx, kubeClient, userWorkspace, []string{consumerUser}, nil, []string{"admin", "access"})
+
+	serviceKcpClient, err := kcpclientset.NewForConfig(framework.UserConfig(providerUser, rest.CopyConfig(cfg)))
+	require.NoError(t, err)
+	serviceDynamicClusterClient, err := kcpdynamic.NewForConfig(framework.UserConfig(providerUser, rest.CopyConfig(cfg)))
+	require.NoError(t, err)
+
+	userKcpClient, err := kcpclientset.NewForConfig(framework.UserConfig(consumerUser, rest.CopyConfig(cfg)))
+	require.NoError(t, err)
+
+	t.Logf("Install APIResourceSchema into service provider workspace %q", serviceWorkspace)
+	serviceProviderKcpClient, err := kcpclientset.NewForConfig(framework.UserConfig(providerUser, rest.CopyConfig(cfg)))
+	require.NoError(t, err)
+	mapper := restmapper.NewDeferredDiscoveryRESTMapper(memory.NewMemCacheClient(serviceProviderKcpClient.Cluster(serviceWorkspace).Discovery()))
+	err = helpers.CreateResourceFromFS(ctx, serviceDynamicClusterClient.Cluster(serviceWorkspace), mapper, nil, "apiresourceschema_cowboys.yaml", testFiles)
+	require.NoError(t, err)
+
+	t.Logf("Get the root scheduling APIExport's identity hash")
+	schedulingAPIExport, err := kcpClient.Cluster(tenancyv1alpha1.RootCluster).ApisV1alpha1().APIExports().Get(ctx, "scheduling.kcp.dev", metav1.GetOptions{})
+	require.NoError(t, err)
+	require.True(t, conditions.IsTrue(schedulingAPIExport, apisv1alpha1.APIExportIdentityValid))
```

Comment:
this needs to be wrapped in a `framework.Eventually` as this is applied eventually.

Reply:
nevermind, this should be part of the bootstrap phase